src/util/widgetrendertimer.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <QObject>
+
+#include "util/duration.h"
+#include "util/timer.h"
+
+// A helper class for rendering widgets on a timer when they need updating.
+// Prevents calling QWidget::update too quickly. Controlled by two parameters:
+// - renderFrequency: The frequency to render the widget at in response
+//   to updates.
+// - inactivityTimeout: The timeout after which the widget's render timer is
+//   deactivated.
+//
+// This class was created in response to Bug #1793015. With Qt 4, we would
```

Comment:
Launchpad Bug #1793015 ;)

Reply:
fine :P. But when we `s/Bug #/lp:/g` then it'll read "Launchpad lp:1793015". I feel like if we stick to the "Bug #1234" that has been the standard way, then it'll be easier to mass change later.